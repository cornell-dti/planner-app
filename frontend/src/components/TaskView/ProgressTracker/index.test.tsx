import React from 'react';
import renderer from 'react-test-renderer';
import { ProgressTracker } from '.';

it('Mobile completed ProgressTracker matches snapshot', () => {
  const tree = renderer.create(
    <ProgressTracker inMobileView allTasksCount={3} completedTasksCount={3} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Mobile incompleted ProgressTracker matches snapshot', () => {
  const tree = renderer.create(
    <ProgressTracker inMobileView allTasksCount={1} completedTasksCount={3} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Desktop completed ProgressTracker matches snapshot', () => {
  const tree = renderer.create(
    <ProgressTracker inMobileView={false} allTasksCount={3} completedTasksCount={3} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Desktop incompleted ProgressTracker matches snapshot', () => {
  const tree = renderer.create(
    <ProgressTracker inMobileView={false} allTasksCount={1} completedTasksCount={3} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
