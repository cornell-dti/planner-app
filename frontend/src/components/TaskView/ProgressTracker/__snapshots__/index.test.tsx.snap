// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop completed ProgressTracker matches snapshot 1`] = `
<div
  className="ProgressTracker DesktopView"
>
  <div
    className="Bear"
    style={
      Object {
        "backgroundImage": "url(happy-bear.png)",
        "margin": "0",
      }
    }
  />
  <div
    className="ProgressIndicator"
    style={
      Object {
        "flexDirection": "column-reverse",
        "margin": "0",
        "width": "32px",
      }
    }
  >
    <div
      className="ProgressText"
      style={
        Object {
          "marginBottom": "8px",
          "width": "70px",
        }
      }
    >
      <div
        className="ProgressCountText"
      >
        3/3
      </div>
      <div
        className="ProgressOtherText"
      >
        tasks complete
      </div>
    </div>
    <div
      className="ProgressBarContainer"
      style={
        Object {
          "marginBottom": "1em",
          "width": "2em",
        }
      }
    >
      <div
        className="ProgressBarIndicator"
        style={
          Object {
            "height": "100%",
            "width": "2em",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Desktop incompleted ProgressTracker matches snapshot 1`] = `
<div
  className="ProgressTracker DesktopView"
>
  <div
    className="Bear"
    style={
      Object {
        "backgroundImage": "url(regular-bear.png)",
        "margin": "0",
      }
    }
  />
  <div
    className="ProgressIndicator"
    style={
      Object {
        "flexDirection": "column-reverse",
        "margin": "0",
        "width": "32px",
      }
    }
  >
    <div
      className="ProgressText"
      style={
        Object {
          "marginBottom": "8px",
          "width": "70px",
        }
      }
    >
      <div
        className="ProgressCountText"
      >
        3/1
      </div>
      <div
        className="ProgressOtherText"
      >
        tasks complete
      </div>
    </div>
    <div
      className="ProgressBarContainer"
      style={
        Object {
          "marginBottom": "1em",
          "width": "2em",
        }
      }
    >
      <div
        className="ProgressBarIndicator"
        style={
          Object {
            "height": "300%",
            "width": "2em",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Mobile completed ProgressTracker matches snapshot 1`] = `
<div
  className="ProgressTracker MobileView"
>
  <div
    className="Bear"
    style={
      Object {
        "backgroundImage": "url(happy-bear.png)",
      }
    }
  />
  <div
    className="ProgressIndicator"
    style={
      Object {
        "height": "32px",
      }
    }
  >
    <div
      className="ProgressText"
      style={Object {}}
    >
      <div
        className="ProgressCountText"
      >
        3/3
      </div>
      <div
        className="ProgressOtherText"
      >
        tasks complete
      </div>
    </div>
    <div
      className="ProgressBarContainer"
      style={
        Object {
          "height": "2em",
          "marginLeft": "1em",
        }
      }
    >
      <div
        className="ProgressBarIndicator"
        style={
          Object {
            "height": "2em",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Mobile incompleted ProgressTracker matches snapshot 1`] = `
<div
  className="ProgressTracker MobileView"
>
  <div
    className="Bear"
    style={
      Object {
        "backgroundImage": "url(regular-bear.png)",
      }
    }
  />
  <div
    className="ProgressIndicator"
    style={
      Object {
        "height": "32px",
      }
    }
  >
    <div
      className="ProgressText"
      style={Object {}}
    >
      <div
        className="ProgressCountText"
      >
        3/1
      </div>
      <div
        className="ProgressOtherText"
      >
        tasks complete
      </div>
    </div>
    <div
      className="ProgressBarContainer"
      style={
        Object {
          "height": "2em",
          "marginLeft": "1em",
        }
      }
    >
      <div
        className="ProgressBarIndicator"
        style={
          Object {
            "height": "2em",
            "width": "300%",
          }
        }
      />
    </div>
  </div>
</div>
`;
